README.adoc

```diff
@@ -91,116 +91,127 @@ image::find-build-scans.png[Find build scans,{thumbnail}]
+
+To see which versions are injected, refer to <<Compatibility>>.
 
 > **Warning**
 > - Maven injection only works if `MAVEN_OPTS` is not configured as a global environment variable.
 
 === Disable auto-injection
 
 Disabling the auto-injection requires that all Gradle Enterprise resources are cleaned up from the agents.
-To achieve this, the `JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_INJECTION` environment variable must be set to `true` and the Gradle or Maven injections must be disabled individually.
+To achieve this, the Gradle or Maven injections must be disabled individually (see the following sections).
 This triggers a cleanup of the resources.
-If the auto-injection should be disabled globally, then remove the `JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_INJECTION` as well.
+If the auto-injection should be disabled globally, then unset the `Enable auto-injection` checkbox as well.
 
 ==== Gradle
 
-To disable the auto-injection for Gradle builds, remove the `JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_PLUGIN_VERSION` environment variable.
+To disable the auto-injection for Gradle builds, remove the `Gradle Enterprise Gradle plugin version` in the `Gradle settings` section of the configuration form.
 
 ==== Maven
 
-To disable the auto-injection for Maven builds, remove the `JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_EXTENSION_VERSION` environment variable.
+To disable the auto-injection for Maven builds, unset the `Enable Gradle Enterprise Maven extension auto-injection` checkbox in the `Maven settings` section of the configuration form.
 
 === Control auto-injection based on node labels
 
-Auto-injection can be enable or disabled based on specific node labels.
-To enable auto-injection only on specific nodes set the `JENKINSGRADLEPLUGIN_GRADLE_INJECTION_ENABLED_NODES` for Gradle or `JENKINSGRADLEPLUGIN_MAVEN_INJECTION_DISABLED_NODES` for Maven to a comma separated list of target labels.
-To disable auto-injection on specific nodes set the `JENKINSGRADLEPLUGIN_GRADLE_INJECTION_DISABLED_NODES` for Gradle or `JENKINSGRADLEPLUGIN_MAVEN_INJECTION_DISABLED_NODES` for Maven to a comma separated list of target labels.
+Auto-injection can be enabled or disabled based on specific node labels.
+To enable auto-injection only on specific nodes add the desired labels to the `Gradle auto-injection enabled nodes` list for Gradle or `Maven auto-injection enabled nodes` for Maven.
+To disable auto-injection on specific nodes add the desired labels to the `Gradle auto-injection disabled nodes` list for gradle or `Maven auto-injection disabled nodes` for Maven.
+
 The disabled labels list will take precedence over the enabled labels list.
 
+=== Disabling/removing Gradle Jenkins plugin
+
+To ensure that all Gradle Enterprise resources are cleaned up from the agents, before disabling/removing the plugin the auto-injection has to be manually disabled. Please, refer to the <<Disable auto-injection>> section for details.
```

Comment:
```suggestion
== Uninstalling Gradle Jenkins plugin

To ensure that all Gradle Enterprise resources are cleaned up from the agents, before uninstalling the plugin the auto-injection has to be manually disabled. Please, refer to the <<Disable auto-injection>> section for details.
```

Please also move that top level after the `Gradle Enterprise integration`. (See my change on heading level too)